### ISSUE TITLE
Test test_key_too_short fails 

### ISSUE THREAD

**Issue by tkdchen** [mpdavis/python-jose#176]
When run tests from master branch within testenv py36-base, the expected error `BadDigestError` is not raised.

```
    @pytest.mark.parametrize("Backend,ExceptionType", _backend_exception_types())
    def test_key_too_short(self, Backend, ExceptionType):
        key = Backend(TOO_SHORT_PRIVATE_KEY, ALGORITHMS.ES512)
        with pytest.raises(ExceptionType):
>           key.sign(b'foo')
E           Failed: DID NOT RAISE <class 'ecdsa.keys.BadDigestError'>

tests/algorithms/test_EC.py:121: Failed
```

Steps to reproduce:

* create a virtual environment
* install tox
* `tox -e py36-base`

**Comment by das-g** [comment]
In NixOS/nixpkgs#89723, this was caused by upgrading [ecdsa](https://pypi.org/project/ecdsa/) from 0.14.1 to 0.15 in https://github.com/NixOS/nixpkgs/pull/89723/commits/81667ef22f9f58272db0ff00ae110dcae084b062.

**Comment by blag** [comment]
I have opened a PR in the `ecdsa` project to resolve this. Once that PR is merged a new version of ecdsa is released, I will update python-jose to use the fixed version (or later) which should resolve this issue. Hang tight until then.

**Comment by blag** [comment]
This is fixed in version 3.2.0, which is now available on PyPI and GitHub. It pins the `ecdsa` package to `< 0.15` to avoid this breakage. I hope to have a better solution available in the next release of python-jose.

Closing. Thank you for reporting this.
